apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/AccountStatisticsService.java

```diff
@@ -0,0 +1,89 @@
+import com.apollocurrency.aplwallet.apl.core.account.model.Account;
+import com.apollocurrency.aplwallet.apl.core.account.service.AccountService;
+import com.apollocurrency.aplwallet.apl.core.app.Blockchain;
+import com.apollocurrency.aplwallet.apl.core.app.Convert2;
+import com.apollocurrency.aplwallet.apl.core.chainid.BlockchainConfig;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Objects;
+import java.util.Random;
+
+/**
+ * Account statistic information.
+ */
+@Slf4j
+@Singleton
+public class AccountStatisticsService {
+    private BlockchainConfig blockchainConfig;
+    private Blockchain blockchain;
+    private AccountService accountService;
+
+    @Inject
+    public AccountStatisticsService(BlockchainConfig blockchainConfig,
+                                    Blockchain blockchain,
+                                    AccountService accountService) {
+        this.blockchainConfig = Objects.requireNonNull(blockchainConfig, "blockchainConfig is NULL");
+        this.blockchain = Objects.requireNonNull(blockchain, "blockchain is NULL");
+        this.accountService = Objects.requireNonNull(accountService,"accountService is NULL");
+    }
+
+    public AccountsCountDto getAccountsStatistic(int numberOfAccounts) {
+        log.trace("start getAccountsStatistic = {}", numberOfAccounts);
+        AccountsCountDto dto = new AccountsCountDto();
+        dto.totalSupply = accountService.getTotalSupply();
+        log.trace("getAccountsStatistic dto.totalSupply= {}", dto.totalSupply);
+        dto.totalNumberOfAccounts = accountService.getTotalNumberOfAccounts();
+        log.trace("getAccountsStatistic dto.totalNumberOfAccounts= {}", dto.totalNumberOfAccounts);
+        dto.numberOfTopAccounts = numberOfAccounts;
+        dto.totalAmountOnTopAccounts = accountService.getTotalAmountOnTopAccounts(numberOfAccounts);
+        log.trace("Found totalAmountOnTopAccounts = {} by numberOfAccounts={}", dto.totalAmountOnTopAccounts, numberOfAccounts);
+        List<Account> topHoldersList = accountService.getTopHolders(numberOfAccounts);
+        log.trace("topHoldersList = [{}]", topHoldersList.size());
+        int index = 0;
+        for (Account account : topHoldersList) {
+            long start = System.currentTimeMillis();
+            log.trace("accountJson START for  account={}, index={}", account.getId(), index);
+            AccountEffectiveBalanceDto accountJson = accountBalance(account, false, blockchain.getHeight());
+            log.trace("accountJson = {}, index={} in '{}' msec, DONE", accountJson, index, System.currentTimeMillis() - start);
+            putAccountNameInfo(accountJson, account.getId(), false);
+            dto.topHolders.add(accountJson);
+            index++;
+        }
+        return dto;
+    }
+
+    private AccountEffectiveBalanceDto accountBalance(Account account, boolean includeEffectiveBalance, int height) {
```

Comment:
You can reuse AccountConverter class with same effect without code duplication.

Reply:
I looked into AccountConverter and it doesn't suite here.
I need only small part of Account information - effectiveBallance+guaranteedBalance, but 'effectiveBallance' or 'ballance' is not used inside AccountConverter or other converters.